app/code/Magento/Catalog/Model/ResourceModel/Product.php

```diff
@@ -505,11 +505,29 @@ public function getProductsIdsBySkus(array $productSkuList)
 
         $result = [];
         foreach ($this->getConnection()->fetchAll($select) as $row) {
-            $result[$row['sku']] = $row['entity_id'];
+            $result[$this->getResultKey($row['sku'], $productSkuList)] = $row['entity_id'];
```

Comment:
Why is not enough to have only `strtolower` for hey?

Reply:
Also, looks like we need to fix `\Magento\Catalog\Model\ResourceModel\Product::getProductsIdsBySkus`